shim.c

```diff
@@ -1821,7 +1818,14 @@ EFI_STATUS shim_verify (void *buffer, UINT32 size)
 	if (status != EFI_SUCCESS)
 		goto done;
 
+	/* Measure the binary into the TPM */
+	tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)buffer, size, sha1hash, 4);
```

Comment:
The binary is already measured to PCR4 in the handle_image() function. Isn't that enough? If not, please explain why in the commit message and also make sure that the README.tpm file is updated to reflect what the code does.

Reply:
I've updated the commit message:
```
Currently the only measurement the shim logs in the TPM is that of the EFI
application it directly loads. However, there are no measurements being taken
of application that are being verified through the shim_lock protocol. In this
patch we extend PCR4 for any binary for which Verify is being called through
the shim_lock protocol.